packages/rest/src/parse-json.ts

```
@@ -16,11 +16,35 @@
 /**
  * Factory to create a reviver function for `JSON.parse` to sanitize keys
  * @param reviver - Reviver function
+ * @param prohibitedKeys - An array of keys to be rejected
  */
-export function sanitizeJsonParse(reviver?: (key: any, value: any) => any) {
+export function sanitizeJsonParse(
+  reviver?: (key: any, value: any) => any,
+  prohibitedKeys?: string[],
+) {
+  prohibitedKeys = [
+    '__proto__',
+    'constructor.prototype',
+    ...(prohibitedKeys ?? []),
+  ];
   return (key: string, value: any) => {
-    if (key === '__proto__')
-      throw new Error('JSON string cannot contain "__proto__" key.');
+    if (key === '__proto__') {
```

I think line 45 covers this.

---

This is to fail fast.